src/ship_gui.cpp

```diff
@@ -44,6 +44,7 @@ void DrawShipImage(const Vehicle *v, const Rect &r, VehicleID selection, EngineI
 	int y = ScaleSpriteTrad(-1) + CenterBounds(r.top, r.bottom, 0);
 
 	seq.Draw(x, y, GetVehiclePalette(v), false);
+	if (v->cargo_cap > 0) DrawCargoIconOverlay(x, y, v->cargo_type);
```

Comment:
Why not pass a full reference to `v` to `DrawCargoIconOverlay()` and check the cargo capacity there as an early return?

Reply:
Then `DrawCargoIconOverlays()` wouldn't be able to call it.